### PR TITLE
Build Snappy with RTTI (and fix MacOS exceptions)

### PR DESCRIPTION
**Pull request by markuspf**
### Scope & Purpose

As an alternative to #17256 we could build snappy with RTTI and just keep our hands off these fragile games with mixing -fno-rtti code with -frtti.

Probably much more robust than the hack in #17256, except that the next update to snappy will break this again...

### CODE REVIEW

File: 3rdParty/snappy/snappy-1.1.9/CMakeLists.txt [hunk 1]
```diff
@@ -52,9 +52,12 @@ if(MSVC)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /EHs-c-")
   add_definitions(-D_HAS_EXCEPTIONS=0)
 
+  # Mixing -fno-rtti with -frtti (the default for C++) leads to unwanted
+  # consequences, in particular with exception handling. Since the snappy
+  # developers will not allow fixes in theri repository, we fix it here.
```
Comment (jsteemann) [inline 3rdParty/snappy/snappy-1.1.9/CMakeLists.txt:57]:
```suggestion
  # developers will not allow fixes in their repository, we fix it here.
```

File: 3rdParty/snappy/snappy-1.1.9/CMakeLists.txt [hunk 2]
```diff
@@ -77,9 +80,12 @@ else(MSVC)
   string(REGEX REPLACE "-fexceptions" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-exceptions")
 
+  # Mixing -fno-rtti with -frtti (the default for C++) leads to unwanted
+  # consequences, in particular with exception handling. Since the snappy
+  # developers will not allow fixes in theri repository, we fix it here.
```
Comment (jsteemann) [inline 3rdParty/snappy/snappy-1.1.9/CMakeLists.txt:85]:
```suggestion
  # developers will not allow fixes in their repository, we fix it here.
```

**Review by jsteemann — APPROVED** [review]
LGTM

**Comment by markuspf** [comment]
Will do backports imminently.

**Pull request merged**